tests/parametric/test_otel_logs.py

```diff
@@ -0,0 +1,631 @@
+                4321,
+            ),
+        ],
+    )
+    def test_otlp_logs_custom_endpoint(
+        self, library_env, endpoint_env, test_agent_otlp_grpc_port, otlp_endpoint_library_env, test_agent, test_library
+    ):
+        """Logs are exported to custom OTLP logs endpoint."""
+        with test_library as library:
+            library.write_log("test_otlp_logs_custom_endpoint", LogLevel.INFO, "test_logger")
+
+        assert (
+            urlparse(library_env[endpoint_env]).port == 4321
+        ), f"Expected port 4321 in {urlparse(library_env[endpoint_env])}"
+        log_payloads = test_agent.wait_for_num_log_payloads(1)
+        assert find_log_record(log_payloads, "test_logger", "test_otlp_logs_custom_endpoint") is not None
+
+
+@features.otel_logs_enabled
+@scenarios.parametric
+class Test_FR06_OTLP_Protocols:
+    """FR06: OTLP Protocol Tests"""
+
+    @pytest.mark.parametrize(
+        "library_env",
+        [
+            {
+                "DD_LOGS_OTEL_ENABLED": "true",
+                "OTEL_EXPORTER_OTLP_PROTOCOL": "http/protobuf",
+                "DD_TRACE_DEBUG": None,
+            },
+            {
+                "DD_LOGS_OTEL_ENABLED": "true",
+                "OTEL_EXPORTER_OTLP_PROTOCOL": "grpc",
+                "DD_TRACE_DEBUG": None,
+            },
+        ],
+        ids=["http_protobuf", "grpc"],
+    )
+    def test_otlp_protocols(self, test_agent, test_library, library_env):
+        """OTLP logs are emitted in expected format."""
+        with test_library as library:
+            library.write_log("test_otlp_protocols", LogLevel.INFO, "test_logger")
+
+        log_payloads = test_agent.wait_for_num_log_payloads(1)
+        assert find_log_record(log_payloads, "test_logger", "test_otlp_protocols") is not None
+
+
+@features.otel_logs_enabled
+@scenarios.parametric
+class Test_FR07_Host_Name:
+    """FR07: Host Name Attribute Tests"""
+
+    @pytest.mark.parametrize(
+        "library_env",
+        [
+            {
+                "DD_LOGS_OTEL_ENABLED": "true",
+                "DD_HOSTNAME": "ddhostname",
+                "DD_TRACE_REPORT_HOSTNAME": "true",
+                "DD_TRACE_DEBUG": None,
+            },
+        ],
+    )
+    def test_hostname_from_dd_hostname(self, test_agent, test_library, library_env):
+        """host.name is set from DD_HOSTNAME."""
+        with test_library as library:
+            library.write_log("test_hostname_from_dd_hostname", LogLevel.INFO, "test_logger")
+
+        log_payloads = test_agent.wait_for_num_log_payloads(1)
+        resource = find_resource(log_payloads, "test_logger", "test_hostname_from_dd_hostname")
+        attrs = find_attributes(resource)
+
+        assert attrs.get("host.name") == "ddhostname"
+
+    @pytest.mark.parametrize(
+        "library_env",
+        [
+            {
+                "DD_LOGS_OTEL_ENABLED": "true",
+                "OTEL_RESOURCE_ATTRIBUTES": "host.name=otelenv-host",
```

Comment:
Two questions:
- Can you also test the resource attributes `datadog.host.name` and `host.id` as individual test cases?
- Can we remove `"DD_TRACE_REPORT_HOSTNAME": "true"`? If this is defined on the OTEL_RESOURCE_ATTRIBUTES, it should work regardless of the report hostname flag, right?